packages/ui/src/components/pages/SetupWallet/NewWallet/BackupWallet.tsx

```diff
@@ -0,0 +1,131 @@
+import JsonFile from 'components/shared/export/JsonFile';
+import QrCode from 'components/shared/export/QrCode';
+import CryptoJS from 'crypto-js';
+import useSetupWallet from 'hooks/wallet/useSetupWallet';
+import { setupWalletActions } from 'redux/slices/setup-wallet';
+import { RootState } from 'redux/store';
+import { Props, TransferableObject, WalletRecoveryMethod } from 'types';
+
+interface BackupInstructionProps extends Props {
+  method: WalletRecoveryMethod;
+}
+
+function BackupInstruction({ method }: BackupInstructionProps): JSX.Element {
+  const { t } = useTranslation();
+  const { password, secretPhrase } = useSelector((state: RootState) => state.setupWallet);
+
+  // Preventing re-create the wallet backup when the component is re-rendered
+  const walletBackup = useMemo(() => {
+    if (!secretPhrase || !password) return;
+
+    const encryptedMnemonic = CryptoJS.AES.encrypt(secretPhrase, password).toString();
+
+    return {
+      encryptedMnemonic,
+    } as WalletBackup;
+  }, [secretPhrase, password]);
+
+  if (!walletBackup) return <></>;
+
+  switch (method) {
+    case WalletRecoveryMethod.SecretRecoveryPhrase:
+      return <BackupSecretRecoveryPhrase secretPhrase={secretPhrase!} title='Secret Recovery Phrase' />;
+    case WalletRecoveryMethod.QrCode:
+      return (
+        <QrCode
+          value={walletBackup}
+          object={TransferableObject.Wallet}
+          title='Downloads QR Code'
+          topInstruction={
+            <p>
+              {t<string>(
+                'Downloads the below QR code image and upload it to the cloud (iCloud, Google Drive, ...), you can restore your wallet by scanning or uploading this QR Code to Coong Wallet later at any time with your wallet password.',
+              )}
+            </p>
+          }
+        />
+      );
+    case WalletRecoveryMethod.JsonFile:
+      return (
+        <JsonFile
+          value={walletBackup}
+          object={TransferableObject.Wallet}
+          title='Downloads JSON File'
+          topInstruction={
+            <p>
+              {t<string>(
+                'Downloads the below JSON file and upload it to the cloud (iCloud, Google Drive, ...), you can restore your wallet by uploading this JSON file to Coong Wallet later at any time with your wallet password.',
+              )}
+            </p>
+          }
+        />
+      );
+    default:
+      return <></>;
+  }
+}
+
+interface BackupWalletProps extends Props {
+  method: WalletRecoveryMethod;
+  resetMethod: () => void;
+}
+
+export default function BackupWallet({ method, resetMethod }: BackupWalletProps): JSX.Element {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { password, secretPhrase } = useSelector((state: RootState) => state.setupWallet);
+  const { setup, loading } = useSetupWallet({ secretPhrase, password });
+  const [checked, setChecked] = useState<boolean>(false);
+
+  useEffectOnce(() => {
+    if (secretPhrase) return;
```

Comment:
Why do we need this early return here? In which case that the `secretPhrase` be present in redux before we render this view? Generally, in every case, we should regenerate the secret phrase whenever users go to this view.

Also we're using `setupWalletActions.setSecretPhrase` in `EnterSecretRecoveryPhrase` view as well, please check to make sure there are no problem if we're using it in both places.